src/app/features/add-workplace/new-select-main-service/new-select-main-service.component.ts

```diff
@@ -70,6 +70,29 @@ export class NewSelectMainServiceComponent extends SelectMainServiceDirective {
   }
 
   public setBackLink(): void {
-    this.backService.setBackLink({ url: [`${this.flow}/your-workplace`] });
+    const route = this.isRegulated ? this.getCQCRegulatedBackLink() : this.getNonCQCRegulatedBackLink();
+    this.backService.setBackLink({ url: [this.flow, route] });
+  }
+
+  private getCQCRegulatedBackLink(): string {
+    if (this.workplaceService.manuallyEnteredWorkplace$.value) {
+      return 'workplace-name-address';
+    } else if (this.workplaceService.locationAddresses$.value.length == 1) {
+      return 'your-workplace';
+    } else if (this.workplaceService.locationAddresses$.value.length > 1) {
+      return 'select-workplace';
+    }
```

Comment:
```suggestion
    if (this.workplaceService.manuallyEnteredWorkplace$.value) {
      return 'workplace-name-address';
    } 
    if (this.workplaceService.locationAddresses$.value.length == 1) {
      return 'your-workplace';
    }
    if (this.workplaceService.locationAddresses$.value.length > 1) {
      return 'select-workplace';
    }
```
I think the if/else statements can be hard to read, this way might be easier